pkg/apis/core/validation/validation.go

```diff
@@ -3054,6 +3054,9 @@ func validateImagePullSecrets(imagePullSecrets []core.LocalObjectReference, fldP
 		if !reflect.DeepEqual(strippedRef, currPullSecret) {
 			allErrors = append(allErrors, field.Invalid(idxPath, currPullSecret, "only name may be set"))
 		}
+		if len(currPullSecret.Name) == 0 {
+			allErrors = append(allErrors, field.Required(idxPath.Child("name"), "name cannot be empty"))
```

Comment:
/hold

standard questions when we want to tighten validation:
* does it prevent a no-op update of existing persisted objects that would fail this new validation? (would block storage migration tasks)
* does it prevent deletion of existing persisted objects with finalizers that would fail this new validation? (update is required to remove finalizers to allow deletion)
* does it prevent controllers from updating status of existing persisted objects that would fail this new validation?

we could consider ratcheting validation:
* apply this tightened validation on create
* apply this tightened validation on update where the old object passed this validation (don't let a valid object become invalid)
* skip this tightened validation on update where the old object fails this validation

c.f. https://github.com/kubernetes/kubernetes/issues/59984

Reply:
> does it prevent a no-op update of existing persisted objects that would fail this new validation? (would block storage migration tasks)

No, this only affects the creation of the object. Updating/Patching this object would already fail in absence of the merge key, so we're just applying the same validation on creation.

> does it prevent deletion of existing persisted objects with finalizers that would fail this new validation? (update is required to remove finalizers to allow deletion)

No, it doesn't prevent deletion. I tested it and the object gets deleted.

> does it prevent controllers from updating status of existing persisted objects that would fail this new validation?

No, at least from my tests it doesn't look like it.

> apply this tightened validation on create

This is the goal of this PR

> apply this tightened validation on update where the old object passed this validation (don't let a valid object become invalid)

The update/patch already fails if somebody has a pod with an imagePullSecrets entry with an empty name, because of the strategic merge patch. (so if an object is valid, aka doesn't have an empty-name entry in the imagePullSecrets list, it will remain valid)

> skip this tightened validation on update where the old object fails this validation

An old object in this state would already fail the strategic merge patch validation because it doesn't have the merge key

Reply:
I added this to the list of known bad or likely incorrect items in pods we could add warnings for in https://github.com/kubernetes/kubernetes/issues/94626

Reply:
/close